src/linear_algebra/nonsingular_inverse.lean

```diff
@@ -274,6 +280,62 @@ calc adjugate A ⬝ A = (Aᵀ ⬝ (adjugate Aᵀ))ᵀ :
   by rw [←adjugate_transpose, ←transpose_mul, transpose_transpose]
 ... = A.det • 1 : by rw [mul_adjugate (Aᵀ), det_transpose, transpose_smul, transpose_one]
 
+lemma det_adjugate_of_cancel {A : matrix n n α}
+  (h : ∀ b, A.det * b = A.det ^ fintype.card n → b = A.det ^ (fintype.card n - 1)) :
+  (adjugate A).det = A.det ^ (fintype.card n - 1) :=
+h (adjugate A).det (calc A.det * (adjugate A).det = (A ⬝ adjugate A).det   : (det_mul _ _).symm
+                                              ... = A.det ^ fintype.card n : by simp [mul_adjugate])
+
+lemma adjugate_eq_one_of_card_eq_one {A : matrix n n α} (h : fintype.card n = 1) : adjugate A = 1 :=
+begin
+  ext i j,
+  have univ_eq_i := univ_eq_singleton_of_card_one i h,
+  have univ_eq_j := univ_eq_singleton_of_card_one j h,
+  have i_eq_j : i = j := singleton_inj.mp (by rw [←univ_eq_i, univ_eq_j]),
+  have perm_eq : (univ : finset (perm n)) = finset.singleton 1 :=
+    univ_eq_singleton_of_card_one (1 : perm n) (by simp [card_univ, fintype.card_perm, h]),
+  simp [adjugate_val, det, univ_eq_i, perm_eq, i_eq_j]
+end
+
+@[simp] lemma adjugate_zero (h : 1 < fintype.card n) : adjugate (0 : matrix n n α) = 0 :=
+begin
+  ext i j,
+  obtain ⟨j', hj'⟩ : ∃ j', j' ≠ j := fintype.exists_ne_of_one_lt_card h j,
+  apply det_eq_zero_of_column_eq_zero j',
+  intro j'',
+  simp [update_column_ne hj']
+end
+
+lemma det_adjugate_eq_one {A : matrix n n α} (h : A.det = 1) : (adjugate A).det = 1 :=
+calc (adjugate A).det
+    = A.det ^ (fintype.card n - 1) : det_adjugate_of_cancel (λ b hb, by simpa [h] using hb)
+... = 1                            : by rw [h, one_pow]
+
+lemma det_adjugate_of_is_unit {A : matrix n n α} (h : is_unit A.det) :
```

Comment:
Add a comment that the assumption `h` is not necessary, but that this would require a different proof.